drivers/power/ltc3350/ltc3350.c

```diff
@@ -0,0 +1,371 @@
+ * 		       parameters.
+ *
+ * @return ret       - Result of the initialization procedure.
+*******************************************************************************/
+int ltc3350_init(struct ltc3350_dev **device,
+		 struct ltc3350_init_param *init_param)
+{
+	struct ltc3350_dev *dev;
+	int ret = 0;
+	dev = (struct ltc3350_dev *)malloc(sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	ret = no_os_i2c_init(&dev->i2c_desc, &init_param->i2c_init);
+
+	if (ret) {
+		no_os_free(dev);
+	} else {
+		*device = dev;
+	}
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Free the resources allocated by ltc3350_init().
+ *
+ * @param dev - The device structure.
+ *
+ * @return ret - Result of the remove procedure.
+*******************************************************************************/
+int ltc3350_remove(struct ltc3350_dev *dev)
+{
+	int ret = 0;
+
+	ret = no_os_i2c_remove(dev->i2c_desc);
+	no_os_free(dev);
+
+	return ret;
+}
+
+/******************************************************************************/
+/***************************************************************************//**
+ * @brief Reads data from the device (I2C).
+ *
+ * @param dev          - The device structure.
+ * @param base_address - Address of the base register.
+ * @param size         - The number of bytes to be read, returned in read_data.
+ * @param read_data    - The read data value
+ *
+ * @return ret         - Result of the reading procedure.
+*******************************************************************************/
+int ltc3350_read_device_data(struct ltc3350_dev *dev, uint8_t base_address,
+			     uint16_t *read_data)
+{
+	int ret = 0;
+	uint8_t rx_buff[2] = {0};
+
+	ret = no_os_i2c_write(dev->i2c_desc, &base_address, 1, 1);
+	if (ret)
+		return ret;
+
+	ret = no_os_i2c_read(dev->i2c_desc, rx_buff, 2, 1);
+
+	if (ret) {
+		*read_data = 0;
+			
+	} else {
+		*read_data = (rx_buff[1] << 8) | rx_buff[0];
+	}
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Writes data to the device (I2C)
+ *
+ * @param dev          - The device structure.
+ * @param base_address - Address of the base register.
+ * @param size         - The number of bytes to be written. It is the size
+ *                       of the write_data value in bytes.
+ * @param write_data   - The data which is going to be written.
+ *
+ * @return ret         - Result of the writing procedure.
+*******************************************************************************/
+int ltc3350_write_device_data(struct ltc3350_dev *dev, uint8_t base_address,
+			      uint16_t write_data)
+{
+	int ret = 0;
+	uint8_t txbuff[3] = {0};
+
+	txbuff[0] = base_address;
+	txbuff[1] = write_data & 0xFF;
+	txbuff[2] = write_data >> 8;
+
+	ret = no_os_i2c_write(dev-> i2c_desc, txbuff, 3, 1);
+
+	return ret;
+}
+
+#if	LTC3350_USE_MEASUREMENTS
```

Comment:
does this macro add any value to the code? If the users want to call the functions guarded by this macro in their application will do it or not, based on their specific needs.

Reply:
Could this setting be added in the initialization parameters?

Reply:
Hello they are added in the header file and have the option of switching them off. I have currently disabled the USE_ALARMS switch as it can be optional. Also, depending on the requirement, the USE_MEASUREMENTS can be optional.

Reply:
> Hello they are added in the header file and have the option of switching them off. I have currently disabled the USE_ALARMS switch as it can be optional. Also, depending on the requirement, the USE_MEASUREMENTS can be optional.

As stated in your reply, "depending on the requirement". I suppose these requirements are based on the application in which the driver/parts is used, right? A driver should expose as much as it can the capabilities of a part. It's user's choice what they Therefore these macro definitions should be placed in the application and guard the function calls, not the function definitions.